src/components/Layout.ts

```diff
@@ -0,0 +1,113 @@
+import { MAP } from "../constants/map";
+import { images } from "../image/preload";
+import { Items } from "../enums/items";
+
+export type MapData = {
+  width: number;
+  height: number;
+  tiles: number[][];
+};
+
+export class Layout {
+  x: number = 0;
+  y: number = 0;
+  mapData: MapData;
+  ctx: CanvasRenderingContext2D;
+  img: HTMLImageElement = images.wall.concreteWall;
+
+  constructor(mapData: MapData, ctx: CanvasRenderingContext2D) {
+    this.mapData = mapData;
+    this.ctx = ctx;
+  }
+
+  /**
+   * Draw layout on canvas
+   */
+  draw() {
+    this.ctx.drawImage(
+      this.img,
+      0,
+      0,
+      15,
```

Comment:
Keep this in constants. Why 15 instead of 16 here?

Reply:
yes, it should be 16. i will put them in constants before merging this PR